Library/Homebrew/tab.rb

```diff
@@ -407,4 +430,151 @@ def to_s
     end
     s.join(" ")
   end
+
+  sig { returns(Hash) }
+  def to_spdx_sbom
+    name = tabfile.dirname.dirname.basename
+    runtime_full = runtime_dependencies.map do |dependency|
+      {
+        "SPDXID"           => "SPDXRef-Package-SPDXRef-#{dependency["full_name"].split("/").last}-#{dependency["version"]}",
+        "name"             => dependency["full_name"].split("/").last,
+        "versionInfo"      => dependency["pkg_version"],
+        "filesAnalyzed"    => false,
+        "licenseDeclared"  => "NOASSERTION",
+        "licenseConcluded" => dependency["license"] || "NOASSERTION",
+        "downloadLocation" => "NOASSERTION",
+        "copyrightText"    => "NOASSERTION",
+        "checksums"        => [],
+        "externalRefs"     => [
+          {
+            "referenceCategory" => "PACKAGE-MANAGER",
+            "referenceLocator"  => "pkg:brew/#{dependency["full_name"]}@#{dependency["version"]}",
```

Comment:
NB: The `brew:` PURL spec is inflight here: https://github.com/package-url/purl-spec/pull/281

Reply:
I believe this to be compliant with that spec